### PR TITLE
[Bug Fix] Correct the monthly summary active metrics

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** Customer created issue. 

**This PR will result in the following new package version:** 0.3.0

No column changes were made, so the model itself didn't changed 

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bug Fixes
- We have modified the logic in our `workday__monthly_summary` to segment out inactive employees within our monthly employee and worker active metrics.
  - Now utilize the `is_active` filter to account for employees who are active at the end of a month. We then filter them into active employee and worker metrics, particularly all `active_*` and `avg_*` fields. 
  - This new logic should now account for employees that move between active and inactive states while still being employed, so that `avg_days_as_employee` and `avg_days_as_worker` are properly calculated.
  - Removed existing conditions for active metrics as they only handled a subset of employee/worker cases, whereas `is_active` appears to provide more comprehensive coverage.

## Under The Hood
- Added consistency and integrity tests within integration tests for `workday__monthly_summary` and `workday__employee_daily_history` to ensure proper validation of model changes. 
- Added `employee_history_enabled: true` variable to `integration_tests/dbt_project.yml` for testing history model changes and generating docs.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [NA] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [NA] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [NA] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Created consistency and integrity tests to look at row count on monthly summary, compare non-active metrics on monthly summary, and employee counts on source and final models.  Ran on development data to confirm. 

Screenshot 2024-10-25 at 8 28 27 AM

### If you had to summarize this PR in an emoji, which would it be?
 
🧑‍🏭

### CODE REVIEW

File: integration_tests/tests/integrity/integrity_employee_count.sql
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/integrity/integrity_employee_count.sql:43]: Does this need to be a `left join` instead of a `join`? When I inspect these results I see that the test is passing, but not actually. 

When I run this in the database but comment out the final `where total_employee_source != total_employee_end` statement I get null records for the `total_employee_source` record.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-avinash thanks for working through this PR. I have a few comments and requests before approval.

File: integration_tests/tests/integrity/integrity_employee_count.sql
Comment (fivetran-avinash) [reply integration_tests/tests/integrity/integrity_employee_count.sql]: Thank you for catching. This integrity test took a lot of time to figure out the proper logic, with the surrogate key not making it easy to compare the source data with the end model.

I have reworked the logic and the test should be passing now.

File: CHANGELOG.md
Comment (fivetran-avinash) [reply CHANGELOG.md]: Updated.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz this is ready for re-review

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

File: CHANGELOG.md
Comment (fivetran-catfritz) [inline CHANGELOG.md:8]:
```suggestion
  - Updated the `workday__monthly_summary` model to segment out inactive employees within the monthly employee and worker active metrics.
    - The model now utilizes the `is_active` field to filter employees who are active at the end of a month, which are further filtered into active employee and worker metrics, particularly all `active_*` and `avg_*` fields. 
    - This new logic accounts for employees that move between active and inactive states while still being employed, so `avg_days_as_employee` and `avg_days_as_worker` are properly calculated.
    - Removed existing conditions for active metrics as they only handled a subset of employee/worker cases, whereas `is_active` appears to provide more comprehensive coverage.
```

File: CHANGELOG.md
Comment (fivetran-catfritz) [inline CHANGELOG.md:7]: Some wording suggestions:
```suggestion
  - Updated the `workday__monthly_summary` model to segment out inactive employees within the monthly employee and worker active metrics.
    - The model now utilizes the `is_active` field to filter employees who are active at the end of a month, which are further filtered into active employee and worker metrics, particularly all `active_*` and `avg_*` fields. 
    - This new logic accounts for employees that move between active and inactive states while still being employed, so `avg_days_as_employee` and `avg_days_as_worker` are properly calculated.
```

File: models/workday_history/workday__monthly_summary.sql
Comment (fivetran-catfritz) [inline models/workday_history/workday__monthly_summary.sql:9]: very minor formatting
```suggestion
        row_number() over (partition by employee_id, source_relation, extract(year from date_day), extract(month from date_day) order by date_day desc) as recent_dom_row,
        sum(case when is_active = true then 1 else 0 end) over (partition by employee_id, source_relation order by date_day asc rows unbounded preceding) as days_employee_active,
        sum(case when is_active = true then 1 else 0 end) over (partition by worker_id, source_relation order by date_day asc rows unbounded preceding) as days_worker_active
```

**Review by fivetran-catfritz — APPROVED** [review]
@fivetran-avinash Made a few wording/formatting suggestions, but otherwise lgtm. Approved!

File: models/workday_history/workday__monthly_summary.sql
Comment (fivetran-avinash) [reply models/workday_history/workday__monthly_summary.sql:9]: nice catch!

**Pull request merged**
